### PR TITLE
backupccl: add planning validation to ALTER BACKUP SCHEDULE, emit results

### PR DESCRIPTION
**Pull request by benbardin**
Release note (enterprise change): Alter Backup Schedule will fail if the new backup statement does not pass planning.

Release justification: Improving reliability on feature launching for 22.2.

### CODE REVIEW

**Review by adityamaru — APPROVED** [review]
LGTM sans adding some tests or running alter schedule tests manually.

File: pkg/ccl/backupccl/testdata/backup-restore/alter-schedule/backup-options [hunk 1]
```diff
@@ -53,22 +53,30 @@ with schedules as (show schedules) select id, command->'backup_statement' from s
 $fullID "BACKUP INTO 'nodelocal://1/example-schedule' WITH revision_history = true, encryption_passphrase = '*****', detached"
 $incID "BACKUP INTO LATEST IN 'nodelocal://1/example-schedule' WITH revision_history = true, encryption_passphrase = '*****', detached"
 
+# Add an incompatible option
+
+exec-sql expect-error-ignore
```
[the remark this answers is not in the document]
Comment (benbardin) [reply pkg/ccl/backupccl/testdata/backup-restore/alter-schedule/backup-options]: oh, neat! it needed a little tweaking but yeah!

File: pkg/ccl/backupccl/alter_backup_schedule.go [hunk 2]
```diff
@@ -190,8 +194,40 @@ func doAlterBackupSchedules(
...
 		}
+
+		if err := emitAlteredSchedule(s.incJob, s.incStmt, resultsCh); err != nil {
+			return err
+		}
+	}
+
+	// Emit the full backup schedule after the incremental.
+	// This matches behavior in CREATE SCHEDULE FOR BACKUP.
+	if err := emitAlteredSchedule(s.fullJob, s.fullStmt, resultsCh); err != nil {
```
[the remark this answers is not in the document]
Comment (benbardin) [reply pkg/ccl/backupccl/alter_backup_schedule.go]: done

**Comment by benbardin** [comment]
Added a test for the Emit functionality, and fixed error regex checking in dd tests!

**Pull request merged**